src/generate.rs

```diff
@@ -142,52 +157,119 @@ pub fn generate_pair(opt: &GenerateOptions, rng: &mut impl Rng) -> (Sequence, Se
+            let pattern = ropey::Rope::from_str(
+                std::str::from_utf8(
+                    &(0..opt.pattern_length)
+                        .map(|_| rand_char(rng))
+                        .collect_vec(),
+                )
+                .unwrap(),
+            );
+            a = Vec::new();
+            let mut bb: Vec<u8> = Vec::new();
+            // fill a
+            for _ in 0..opt.length / opt.pattern_length {
+                let mut mutated_pattern = pattern.clone();
+                for _ in 0..(opt.error_rate * opt.pattern_length as f32).ceil() as usize {
+                    make_mutation(&mut mutated_pattern, rng);
                 }
-                b.insert(b_initial.len_bytes() * i, &b_substitutional.to_string());
-                b_initial = b_substitutional.clone();
+                a.append(&mut mutated_pattern.to_string().into_bytes());
             }
-        }
-        ErrorModel::RecursiveMutation => {
-            let mut b_substitutional = b.clone();
-            for i in 0..num_mutations {
-                for _ in 0..num_mutations {
-                    make_mutation(&mut b_substitutional, rng);
+            // fill b
+            for _ in 0..opt.length / opt.pattern_length {
+                let mut mutated_pattern = pattern.clone();
+                for _ in 0..(opt.error_rate * opt.pattern_length as f32).ceil() as usize {
+                    make_mutation(&mut mutated_pattern, rng);
                 }
-                b.insert(
-                    b_substitutional.len_bytes() * i,
-                    &b_substitutional.to_string(),
-                );
+                bb.append(&mut mutated_pattern.to_string().into_bytes());
+            }
+            b = ropey::Rope::from_str(std::str::from_utf8(&bb).unwrap());
+        }
+        ErrorModel::Repeat => {
+            let pattern = ropey::Rope::from_str(
+                std::str::from_utf8(
+                    &(0..opt.pattern_length)
+                        .map(|_| rand_char(rng))
+                        .collect_vec(),
+                )
+                .unwrap(),
+            );
+            a = Vec::new();
+            // fill a
+            for _ in 0..opt.length / opt.pattern_length {
+                a.append(&mut pattern.to_string().into_bytes());
+            }
+            let mut bb = ropey::Rope::from_str(std::str::from_utf8(&a).unwrap());
+            for _ in 0..(opt.length as f32 * opt.error_rate) as usize {
+                make_mutation(&mut bb, rng);
+            }
+            b = bb;
+        }
+        ErrorModel::MutatedRepeat => {
+            let pattern = ropey::Rope::from_str(
+                std::str::from_utf8(
+                    &(0..opt.pattern_length)
+                        .map(|_| rand_char(rng))
+                        .collect_vec(),
+                )
+                .unwrap(),
+            );
+            let mut aa = ropey::Rope::from_str(std::str::from_utf8(&a).unwrap());
```

Comment:
rename to `a_rope`, and initialize with an empty `Rope` instead of copying the existing `a`